### DISCUSSION TITLE
Shutdown

### DISCUSSION THREAD

**Discussion by bdronneau** [drakkan/sftpgo#971] [Q&A] [answered]
Hey,

Thank to provide this wonderful application.
I notice when received SIGTERM, process do not wait pending copy to finish (SFTP PUT for example). There is way to wait the end of a transaction ?

**Comment by drakkan** [comment]
Hi,

there is currently no way. Waiting for active connections to end may take a long time (think about a client uploading a large file over a slow connection). You could enable atomic uploads in you don't want partial files (but this way the client cannot resume a failed upload)

**Comment by bdronneau** [comment]
Hi,

Thank for the feedback. I find a "trick" in order to say "okay no more people". I curl metrics endrpoint and check current log user.
Since I'm using SFTP for backup my local stuff, I put a short idle_timeout in order to not wait too much time.

**Comment by drakkan** [comment, marked as answer]
> Hey,
> 
> Thank to provide this wonderful application. I notice when received SIGTERM, process do not wait pending copy to finish (SFTP PUT for example). There is way to wait the end of a transaction ?

https://github.com/drakkan/sftpgo/commit/db0e58ae7ed7143f3c5c2ecd9009d82cf0b0c844

**Reply by bdronneau** [reply]
Nice, I'll give a try next week.
